onnxruntime/contrib_ops/cpu/gather_nd.cc

```diff
@@ -110,8 +110,8 @@ Status GatherNDString<Tind>::Compute(OpKernelContext* context) const {
   Prepare p;
   ONNXRUNTIME_RETURN_IF_ERROR(PrepareForCompute<Tind>(context, p));
 #pragma omp parallel for
-  for (size_t i = 0; i < p.element_offsets.size(); ++i) {
-    for (size_t j = 0; j < p.element_to_copy; ++j) {
+  for (int64_t i = 0; i < static_cast<int64_t>(p.element_offsets.size()); ++i) {
```

Comment:
Does it work on 32 bits systems?

Reply:
Good question!
On 32 bit os - size() has 4 bytes while on 64 it is 8, right?
Either way, conversion to int64_t is valid by cpp standard.
The only loophole here may be that when size() has 8 bytes and it is big enough, we may receive a wrong int64_t after cast, however, the situation should only exist theoretically...

Reply:
FYI, https://en.cppreference.com/w/cpp/language/implicit_conversion

Reply:
Is it ok to use int64_t as loop index on 32 bits system, for openmp?

Reply:
Should be fine as long as it is supported cpp primitive integer type - believe it is all compile's concern, not coder.